pkg/controller/s3/bucket/replicationConfig.go

```diff
@@ -222,3 +218,14 @@ func (in *ReplicationConfigurationClient) Delete(ctx context.Context, bucket *v1
 	).Send(ctx)
 	return errors.Wrap(err, replicationDeleteFailed)
 }
+
+func sortReplicationRules(rules []awss3.ReplicationRule) {
+	if rules == nil {
+		return
+	}
+	for i := range rules {
+		if rules[i].Filter != nil && rules[i].Filter.And != nil {
+			s3.SortS3TagSet(rules[i].Filter.And.Tags)
+		}
+	}
```

Comment:
```suggestion
	for i := range rules {
		if rules[i].Filter != nil && rules[i].Filter.And != nil {
			s3.SortS3TagSet(rules[i].Filter.And.Tags)
		}
	}
```

That was what I meant. `for` loop won't run at all if `rules == nil` or `len(rules) == 0`. So, we don't need that check there.